src/network/peers.rs

```diff
@@ -581,6 +581,19 @@ impl<N: Network, E: Environment> Peers<N, E> {
             Some((_, outbound)) => {
                 // Ensure sufficient time has passed before needing to send the message.
                 let is_ready_to_send = match message {
+                    Message::Ping(_, _, _, _, ref mut data) => {
+                        let block_header = if let Data::Object(block_header) = data {
+                            block_header
+                        } else {
+                            panic!("Logic error: the block header shouldn't have been serialized yet.");
+                        };
+
+                        // Perform non-blocking serialisation of the block header.
+                        let serialized_header = bincode::serialize(&block_header).expect("Block header serialization is bugged");
```

Comment:
if I'm not missing anything, this call is blocking and you should use `Data::serialize` here instead

Reply:
Yep, good catch 👍

Reply:
I used the same pattern as for the other messages, which I beleive to be correct here? 🤔

Reply:
I don't have a good way of double-checking this right now, but if my hunch is correct it will be easy to tweak afterwards, along with the other case where this is done in the same fashion; this change does make serialization happen ahead of time, though, so it's a step in the right direction :ok_hand:.

Reply:
We effectively unwrap the header from the `Data` wrapper, I think all we'd need is not to do that and use the `Data::serialise` method instead?

Reply:
Yeah, that's what I was thinking, as it's `async`, no?

Reply:
Should work, I'll file a follow-up to keep the change self-contained 👍

Reply:
Filed #1357.